Azure/sonic-buildimage
# [dhcpmon] Monitor Mgmt Interface For DHCP Packets

When BGP routes are missing, DHCP packets get relayed over mgmt
interface. This results in dhcpmon alerting that DHCP packets are
not being relayed. This is PR include mgmt interface as uplink
device, and so, if DHCP packet gets relayed over mgmt interface,
regular dhcpmon alert will not be issues. Instead, dhcpmon will
check the mgmt interface counts and issue a separate alert regarding
packets travelling through mgmt network.

In addition, this PR include the following enhancements:
1. Add SIGUSR1 handler that prints out current packet counts
2. Increase alert grace window to 3 minutes from current 2 min
3. Time is not computed more accurately
4. Print vlan name before counters

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

**- Why I did it**
DHCP Monitor is alerting when DHCP packets got over mgmt network

**- How I did it**
1. Monitored mgmt interface
Also, Added
2. Add SIGUSR1 handler that prints out current packet counts
3. Increase alert grace window to 3 minutes from currently 2 minutes
4. Time is now computed more accurately
5. Print vlan name before counters

**- How to verify it**
Bring down BGP session, and noticed no dhcpmon alert get triggered. Instead the following trigger is observed:
```
Sep  6 07:21:35.928763 str-s6000-acs-14 ALERT dhcp_relay#dhcpmon[43]: dhcpmon detected DHCP packets traveling through mgmt interface (please check BGP routes.) Duration: 234 (sec) for intf: 'eth0'
Sep  6 07:21:35.928763 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [            eth0] DHCP Discover rx/tx:          0/       288, Offer rx/tx:          0/         0, Request rx/tx:          0/         0, ACK rx/tx:          0/         0
Sep  6 07:22:29.927983 str-s6000-acs-14 ALERT dhcp_relay#dhcpmon[43]: dhcpmon detected DHCP packets traveling through mgmt interface (please check BGP routes.) Duration: 288 (sec) for intf: 'eth0'
Sep  6 07:22:29.927983 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [            eth0] DHCP Discover rx/tx:          0/       336, Offer rx/tx:          0/         0, Request rx/tx:          0/         0, ACK rx/tx:          0/         0
```
New signal handler for SIGUSR1
```
Sep  6 07:21:48.670477 str-s6000-acs-14 ALERT dhcp_relay#dhcpmon[43]: Received signal: 'User defined signal 1'
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [            eth0] DHCP Discover rx/tx:          0/       288, Offer rx/tx:          0/         0, Request rx/tx:          0/         0, ACK rx/tx:          0/         0
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [ PortChannel0004] DHCP Discover rx/tx:          0/        55, Offer rx/tx:          0/         0, Request rx/tx:          0/        55, ACK rx/tx:          0/         0
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [ PortChannel0003] DHCP Discover rx/tx:          0/        40, Offer rx/tx:          0/         0, Request rx/tx:          0/        40, ACK rx/tx:          0/         0
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [ PortChannel0002] DHCP Discover rx/tx:          0/        70, Offer rx/tx:          5/         0, Request rx/tx:          0/        70, ACK rx/tx:          5/         0
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [ PortChannel0001] DHCP Discover rx/tx:          0/        75, Offer rx/tx:          0/         0, Request rx/tx:          0/        75, ACK rx/tx:          0/         0
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [        Vlan1000] DHCP Discover rx/tx:         11/         0, Offer rx/tx:          0/         5, Request rx/tx:          5/         0, ACK rx/tx:          0/         5
Sep  6 07:21:48.670477 str-s6000-acs-14 NOTICE dhcp_relay#dhcpmon[43]: [    Agg-Vlan1000] DHCP Discover rx/tx:         11/       528, Offer rx/tx:          5/         5, Request rx/tx:          5/       240, ACK rx/tx:          5/         5
```

**- Which release branch to backport (provide reason below if selected)**


**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
